pkg/utils/validator/clusterresourceoverride_test.go

```diff
@@ -0,0 +1,187 @@
+			overrideCount: 0,
+			operation:     admissionv1.Create,
+			want:          true,
+		},
+		"create override with less than 100 overrides": {
+			overrideCount: 99,
+			operation:     admissionv1.Create,
+			want:          true,
+		},
+		"create override with exactly 100 overrides": {
+			overrideCount: 100,
+			operation:     admissionv1.Create,
+			want:          false,
+		},
+		"update override with exactly 100 overrides": {
+			overrideCount: 100,
+			operation:     admissionv1.Update,
+			want:          true,
+		},
+	}
+
+	// Run the tests
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			croList := &fleetv1alpha1.ClusterResourceOverrideList{}
+			for i := 0; i < tt.overrideCount; i++ {
+				cro := fleetv1alpha1.ClusterResourceOverride{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: fmt.Sprintf("override-%d", i),
+					},
+				}
+				croList.Items = append(croList.Items, cro)
+			}
+			if got := ValidateClusterResourceOverrideLimit(tt.operation, croList); got != tt.want {
+				t.Errorf("ValidateClusterResourceOverrideLimit() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestValidateCROResourceLimit(t *testing.T) {
+	tests := map[string]struct {
+		cro  fleetv1alpha1.ClusterResourceOverride
+		want bool
+	}{
+		// TODO: Add test cases.
+		"create one cluster resource override for resource foo": {
+			cro: fleetv1alpha1.ClusterResourceOverride{
+				Spec: fleetv1alpha1.ClusterResourceOverrideSpec{
+					ClusterResourceSelectors: []fleetv1beta1.ClusterResourceSelector{
+						{
+							Name: "foo",
+						},
+					},
+				},
+			},
+			want: true,
+		},
+		"one override, multiple selectors for 1 existing cluster override": {
+			cro: fleetv1alpha1.ClusterResourceOverride{
+				Spec: fleetv1alpha1.ClusterResourceOverrideSpec{
+					ClusterResourceSelectors: []fleetv1beta1.ClusterResourceSelector{
+						{
+							Name: "example-1",
+						},
+						{
+							Name: "bar",
+						},
+					},
+				},
+			},
+			want: false,
+		},
+		"one override, multiple selectors for existing cluster overrides": {
+			cro: fleetv1alpha1.ClusterResourceOverride{
+				Spec: fleetv1alpha1.ClusterResourceOverrideSpec{
+					ClusterResourceSelectors: []fleetv1beta1.ClusterResourceSelector{
+						{
+							Name: "example-1",
+						},
+						{
+							Name: "example-2",
+						},
+					},
+				},
+			},
+			want: false,
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			croList := &fleetv1alpha1.ClusterResourceOverrideList{}
+			for i := 0; i < 3; i++ {
+				cro := fleetv1alpha1.ClusterResourceOverride{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: fmt.Sprintf("override-%d", i),
+					},
+					Spec: fleetv1alpha1.ClusterResourceOverrideSpec{
+						ClusterResourceSelectors: []fleetv1beta1.ClusterResourceSelector{
+							{
+								Name: fmt.Sprintf("example-%d", i),
+							},
+						},
+					},
+				}
+				croList.Items = append(croList.Items, cro)
+			}
```

Comment:
can move this list before the line 164. we don't need to build the inputs for each tests as you only read these data and won't write them.